sympy/ntheory/residue_ntheory.py

```diff
@@ -742,6 +742,63 @@ def _nthroot_mod1(s, q, p, all_roots):
         return res
     return min(res)
 
+def _nthroot_mod_composite(a, n, p):
+    """
+    Find the solutions to ``x**n = a mod p`` when p is not prime.
+    """
+    f = factorint(p)
+    dd = {}
+    for b, e in f.items():
+        tot_roots = set()
+        if e == 1:
+            root = nthroot_mod(a, n, b, True)
```

Comment:
```suggestion
            root = nthroot_mod(a, n, b, True) or []
```

Reply:
(and then the next two lines can be deleted)

Reply:
yes, and one more thing if #18212 is resolved we can delete 
` if (pow(b, n) - a) % b == 0:
         roots.append(b)`